### PR TITLE
Created userbook module

### PR DESCRIPTION
**Pull request by Xungerrrr**

### CODE REVIEW

File: server/userbook/userbook.schema.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import { Document } from 'mongoose';
+
+export type BookDocument = UserBook & Document;
+
+@Schema()
+export class UserBook {
+
+  @Prop()
+  // book id
+  bid: string;
```
Comment (hans-lizihan) [inline server/userbook/userbook.schema.ts:11]:
```suggestion
  book_id: string;
```

File: server/userbook/userbook.schema.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  @Prop()
+  // user id
+  uid: string;
```
Comment (hans-lizihan) [inline server/userbook/userbook.schema.ts:15]:
```suggestion
  user_id: string;
```

File: server/userbook/userbook.schema.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import { Document } from 'mongoose';
+
+export type BookDocument = UserBook & Document;
```
Comment (hans-lizihan) [inline server/userbook/userbook.schema.ts:4]:
```suggestion
export type UserBookDocument = UserBook & Document;
```

File: server/userbook/userbook.schema.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+@Schema()
+export class UserBook {
+
+  @Prop()
+  // book id
+  bid: string;
+
+  @Prop()
```
Comment (hans-lizihan) [inline server/userbook/userbook.schema.ts:13]:
```suggestion
  @Prop({ type: [{ type: mongoose.Schema.Types.ObjectId, ref: 'User' }] })
```

File: server/userbook/userbook.schema.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import { Document } from 'mongoose';
+
+export type BookDocument = UserBook & Document;
+
+@Schema()
+export class UserBook {
+
+  @Prop()
```
Comment (hans-lizihan) [inline server/userbook/userbook.schema.ts:9]:
```suggestion
  @Prop({ type: [{ type: mongoose.Schema.Types.ObjectId, ref: 'User' }] })
```

File: server/userbook/userbook.schema.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  // user id
+  uid: string;
+
+  @Prop({
+    enum: ['READING', 'WANT_TO_READ', 'READ'],
+  })
+  // bookshelf
+  bookshelf: string;
```
Comment (hans-lizihan) [inline server/userbook/userbook.schema.ts:21]: could be a enum as well, rather than string

Comment (Xungerrrr) [reply server/userbook/userbook.schema.ts]: Updated. Please review the changes. Thanks.

**Review by hans-lizihan — APPROVED** [review]

**Pull request merged**